### PR TITLE
Add more examples to the documentation

### PR DESCRIPTION
**Pull request by flobernd**
Adds copy/paste-able examples for very common use-cases.

### CODE REVIEW

**Review by marciw — COMMENTED** [review]
Thank you @flobernd ! 

Small edits for Elastic docs style for headings.

One thought: would it make sense to have an `examples` subfolder under `usage`  to collect the 4 pages? Not sure...

Am also going to add a commit to add the pages to the nav

File: docs/usage/index.asciidoc
Comment (marciw) [inline docs/usage/index.asciidoc:1]: Adds the examples to the left nav

**Comment by flobernd** [comment]
Thanks for the review and the suggestions @marciw 🙂 

> One thought: would it make sense to have an `examples` subfolder under `usage` to collect the 4 pages? Not sure...

Could definitely make sense - especially when we add even more examples.

File: docs/usage/aggregations.asciidoc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			Field = Infer.Field<Person>(x => x.Age)
+		})}
+	},
+	Size = 10
+});
+----
+
+[discrete]
+==== Consume the response
+==== Consume the response
```
Comment (marciw) [inline docs/usage/aggregations.asciidoc:52]:
```suggestion
```

File: docs/usage/aggregations.asciidoc [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	Aggregations = new Dictionary<string, Aggregation>
+	{
+		{ "firstnames", topLevelAggregation}
+	},
+	Size = 10
+});
+----
+
+[discrete]
+==== Consuming the Response
```
Comment (marciw) [inline docs/usage/aggregations.asciidoc:122]:
```suggestion
```

**Review by marciw — COMMENTED** [review]
Somehow we ended up with a couple of duplicate headings -- going to accept these suggestions to fix

**Review by marciw — APPROVED** [review]
LGTM!

**Comment by flobernd** [comment]
Thank you again @marciw!

Seems I somehow messed up the rebase which caused the headers to get duplicated.

**Pull request merged**